hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -596,15 +654,25 @@ def _batch_spec(self):
     async def _open_batch(self) -> Batch:
         batch_spec = self._batch_spec()
         batch_json = await (await self._client._post('/api/v1alpha/batches/create', json=batch_spec)).json()
+        self._update_id = batch_json['update_id']
         return Batch(self._client,
                      batch_json['id'],
                      self.attributes,
-                     batch_spec['n_jobs'],
                      self.token,
                      submission_info=BatchSubmissionInfo(False))
 
-    async def _close_batch(self, id: int):
-        await self._client._patch(f'/api/v1alpha/batches/{id}/close')
+    def _update_spec(self) -> dict:
+        update_token = secrets.token_urlsafe(32)
+        return {'n_jobs': len(self._jobs), 'token': update_token}
+
+    async def _create_update(self, batch_id: int) -> int:
+        update_spec = self._update_spec()
+        update_json = await (await self._client._post(f'/api/v1alpha/batches/{batch_id}/updates/create', json=update_spec)).json()
+        return int(update_json['update_id'])
+
+    async def _commit_update(self, batch_id: int, update_id: int) -> int:
+        commit_json = await (await self._client._patch(f'/api/v1alpha/batches/{batch_id}/updates/{update_id}/commit')).json()
+        return int(commit_json['start_job_id'])
```

Comment:
I originally had the commit return the `start_job_id` because it's not actually needed by the client until the update is committed. It might have been premature optimization on my part hiding that information from the client until the update is committed, and it is probably clearer to provide it in the create update endpoint along with the update id, so I'm fine with your version. But this is probably the cause of some of the errors that it's currently hitting.